cli/internal/runsummary/run_summary.go

```diff
@@ -16,7 +25,19 @@ type RunSummary struct {
 	TurboVersion      string             `json:"turboVersion"`
 	GlobalHashSummary *GlobalHashSummary `json:"globalHashSummary"`
 	Packages          []string           `json:"packages"`
-	Tasks             []TaskSummary      `json:"tasks"`
+	Tasks             *[]*TaskSummary    `json:"tasks"`
+}
+
+// Save saves the run summary to a file
+func (summary *RunSummary) Save(singlePackage bool) error {
+	json, err := summary.FormatJSON(singlePackage)
+	if err != nil {
+		return err
+	}
+
+	nonce := "abcdef"
+	filename := fmt.Sprintf("vercel-turbo-run-%s-%s.json", nonce, time.Now().Format(time.RFC3339))
```

Comment:
We need to enable this to be shared without collision. We should probably consider using a KSUID:

https://github.com/segmentio/ksuid

Reply:
Read through [segments blog post](https://segment.com/blog/a-brief-history-of-the-uuid/) about KSUIDs, very interesting! Good with me to use it, although I'm opting to keep the timestamp as well so it's a tad easier to differentiate which file is which